maven/templates/rules.bzl

```diff
@@ -423,6 +440,60 @@ assemble_maven = rule(
 )
 
 
+###############################
+####        JAVADOC        ####
+###############################
+
+def _javadoc(ctx):
+    target = ctx.attr.target
+
+    target_string = target[JavaLibInfo].target_coordinates.to_list()[-1]
+    maven_coordinates = _parse_maven_coordinates(target_string)
+
+    output_jar = ctx.actions.declare_file("{}:{}-javadoc.jar".format(maven_coordinates.group_id, maven_coordinates.artifact_id))
+
+    # extract all files
+    src_files = []
+    for src in target[JavaLibInfo].srcs.to_list():
+        src_files += src.files.to_list()
+
+    # extract paths to use in the javadoc command
+    src_list = []
+    for src in src_files:
+        src_list += [src.path]
+
+    # https://docs.oracle.com/en/java/javase/11/javadoc/javadoc-command.html#GUID-B0079316-8AA3-475B-8276-6A4095B5186A
+    cmd = [
+        "mkdir -p %s" % maven_coordinates.artifact_id,
+        "javadoc -d %s %s" % (maven_coordinates.artifact_id, " ".join(src_list)),
```

Comment:
This does not look like the correct way to achieve javadoc generation, considering that it does not use the bazel java toolchain and goes for the system one instead, which may break if the bazel toolchain is a different version. Have a look at this: https://github.com/GerritCodeReview/bazlets/commit/13330934a0e577d60719551ffb746f64b0bb9637 here they get the definite location of the java toolchain being used and use the corresponding javadoc tool directly.

Reply:
Thanks for the review @adammitchelldev .
I updated the PR to call the javadoc provided by the bazel java toolchain.